src/base/bittorrent/session.cpp

```diff
@@ -2408,6 +2410,31 @@ void Session::saveResumeData()
     }
 }
 
+void Session::saveTorrentsQueue()
+{
+    QMap<int, QString> queue; // Use QMap since it should be ordered by key
+    for (const TorrentHandle *torrent : torrents()) {
+        // We require actual (non-cached) queue position here!
+        const int queuePos = torrent->nativeHandle().queue_position();
+        if (queuePos >= 0)
+            queue[queuePos] = torrent->hash();
+    }
+
+    QByteArray data;
+    for (const QString &hash : queue)
+        data += (static_cast<QString>(hash).toLatin1() + '\n');
```

Comment:
Why the static_cast? `hash` is already of type `QString` here. What am I missing?

Reply:
Several alterations with a lack of free time lead to such results.